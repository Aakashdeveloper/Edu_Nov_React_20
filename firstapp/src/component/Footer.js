import React from 'react'

const Footer = (props) => {
    return(
        <React.Fragment>
            <hr/>
            <center>
                <h2>&copy; Developer Funnel {props.year} {props.month}</h2>
            </center>
        </React.Fragment>
    )
}

export default Footer;