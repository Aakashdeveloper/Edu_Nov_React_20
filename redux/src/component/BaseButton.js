import React,{Component} from 'react';

class BaseButton extends Component {
    constructor(){
        super()

        this.state={
            name:''
        }
    }

    onClick=()=>{
        this.setState({name:'Some Thing else'})
    }
    
    render(){
        return(
            <div>
                <button className="btn btn-success" onClick={this.onClick}>
                    {this.state.name}
                </button>
            </div>
        )
    }
}

export default BaseButton