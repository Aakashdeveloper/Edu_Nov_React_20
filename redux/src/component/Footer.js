import React from 'react';

const Footer = () => {
    return(
        <footer>
            <div>
                <hr/>
                &copy; Developer Funnel
            </div>
        </footer>
    )
}

export default Footer;