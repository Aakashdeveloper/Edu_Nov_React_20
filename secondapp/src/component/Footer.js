import React from 'react';

const Footer = () => {
    return(
        <div>
            <hr/>
            <center>
                <h2>&copy; Edureka 2020</h2>
            </center>
        </div>
    )
}

export default Footer;