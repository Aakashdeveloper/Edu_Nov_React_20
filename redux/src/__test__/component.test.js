import React from 'react';
import {create} from 'react-test-renderer';
import BaseButton from '../component/BaseButton';
import Footer from '../component/Footer';

describe('Functional Testing',() => {
    test('button test',()=>{
        const component = create(<BaseButton/>);
        const instance = component.root;
        const button = instance.findByType("button");
        button.props.onClick();
        expect(button.props.children).toBe("Some Thing else")
    })
    test('Snapshot Testing',() => {
        let tree = create(<Footer/>);
        expect(tree.toJSON()).toMatchSnapshot()
    })
})