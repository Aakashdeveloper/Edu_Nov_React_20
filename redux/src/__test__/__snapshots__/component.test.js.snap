// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Functional Testing Snapshot Testing 1`] = `
<footer>
  <div>
    <hr />
    Â© Developer Funnels
  </div>
</footer>
`;
